sympy/functions/elementary/tests/test_exponential.py

```diff
@@ -570,17 +570,43 @@ def test_polar():
     assert exp_polar(0).is_rational is True  # issue 8008
 
 
+def test_exp_summation():
+    w = symbols("w")
+    m, n, i, j = symbols("m n i j")
+    expr = exp(Sum(w*i, (i, 0, n), (j, 0, m)))
+    assert expr.expand() == Product(exp(w*i), (i, 0, n), (j, 0, m))
+
+
 def test_log_product():
     from sympy.abc import n, m
+    from sympy.concrete import Product
+
     i, j = symbols('i,j', positive=True, integer=True)
     x, y = symbols('x,y', positive=True)
-    from sympy.concrete import Product
-    assert simplify(log(Product(x**i, (i, 1, n)))) == log(Product(x**i, (i, 1, n)))
-    assert simplify(log(Product(x**i*y**j, (i, 1, n), (j, 1, m)))) == \
-            log(Product(x**i*y**j, (i, 1, n), (j, 1, m)))
+    z = symbols('z', real=True)
+    w = symbols('w')
+
+    expr = log(Product(x**i, (i, 1, n)))
+    assert simplify(expr) == expr
+    assert expr.expand() == Sum(i*log(x), (i, 1, n))
+    expr = log(Product(x**i*y**j, (i, 1, n), (j, 1, m)))
+    assert simplify(expr) == expr
+    assert expr.expand() == Sum(i*log(x) + j*log(y), (i, 1, n), (j, 1, m))
 
     expr = log(Product(-2, (n, 0, 4)))
     assert simplify(expr) == expr
+    assert expr.expand() == expr
+    assert expr.expand(force=True) == Sum(log(-2), (n, 0, 4))
+
+    expr = log(Product(exp(z*i), (i, 0, n)))
+    assert expr.expand() == Sum(z*i, (i, 0, n))
+
+    expr = log(Product(exp(w*i), (i, 0, n)))
+    assert expr.expand() == expr
+    assert expr.expand(force=True) == Sum(w*i, (i, 0, n))
+
+    expr = log(Product(i**2*abs(j), (i, 0, n), (j, 0, m)))
+    assert expr.expand() == Sum(2*log(i) + log(j), (i, 0, n), (j, 0, m))
```

Comment:
One of these expression evaluates to zoo and the other to nan:
```julia
In [11]: e1 = log(Product(i**2*abs(j), (i, 0, n), (j, 0, m)))                                                                     

In [12]: e2 = Sum(2*log(i) + log(j), (i, 0, n), (j, 0, m))                                                                        

In [13]: e1.subs(n, 3).subs(m, 3).doit()                                                                                          
Out[13]: zoo

In [14]: e2.subs(n, 3).subs(m, 3).doit()                                                                                          
Out[14]: nan
```
Is this a valid transformation?

Reply:
I think this example only expands because i and j are declared as positive which means that the Product.function.is_positive even though those assumptions don't match the limits of the product in which both indices go from 0.

Reply:
It's a problem with the assumptions. `i` is declared as positive integer, so it assumes that it's non-zero, even if the product starts from zero.

It's probably an unrelated issue.